scripts/build_test.py

```diff
@@ -721,4 +723,169 @@ def mock_require_compiled_js_dir_to_be_valid():
                 self.assertFalse(
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
+    def test_generate_app_yaml(self):
+        temp_file = tempfile.NamedTemporaryFile()
+        temp_file.name = 'app_dev.yaml'
```

Comment:
Better to give this a different name (like mock_app_dev.yaml) to avoid confusion with the existing app_dev.yaml file, otherwise how do you know that the code is editing your tempfile and not the existing file? Hard to keep track.

Reply:
Done